### DISCUSSION TITLE
Handle scrolling when router outlet is inside "overflow: scroll"

### DISCUSSION THREAD

**Discussion by orenelbaum** [solidjs/solid-router#226] [Ideas]
When the `<Routes>` component is inside an element with `overflow: scroll` it messes up both the scroll to top behavior built into the router and the scroll restoration built into the browser. I'm suggesting adding a feature that handles this.
You can see a demonstration of this here:
https://stackblitz.com/github/orenelbaum/solid-router-overflow-scroll-demo
https://github.com/orenelbaum/solid-router-overflow-scroll-demo

**Comment by Brendan-csel** [comment]
React router also only seems to handle restoring the `window` scroll position automatically too - looking at [this code](https://github.com/remix-run/react-router/blob/11156ac7f3d7c1c557c67cc449ecbf9bd5c6a4ca/packages/react-router-dom/index.tsx#L1154-L1158) anyway.

I suspect any other scrolling containers might need to be handled by app code.

**Reply by orenelbaum** [reply]
The code you're looking at is for async rendering of the page which breaks the scroll restoration built into the browser.
There's an open issue for it https://github.com/solidjs/solid-router/issues/99
Why can't we provide the functionality for scrolling containers though? I mean I'm not sure if we can completely automate it but even if we can't we can probably provide the functionality and let the user just do some basic setup instead of making the user either implement it themselves which is harder than it might seem, or go for a 3rd party solution.

**Reply by Brendan-csel** [reply]
Have you seen an API for restoring other container(s) in any other routers?  It would be good to see if there is a proven approach out there.

**Reply by Brendan-csel** [reply]
I see a [pull request](https://github.com/remix-run/react-router/pull/9573) (not merged) in react router to be able to provide an `id` prop to their scroll restoration hook/component for the scrollable container.  So I guess that might be an option - if we add a scroll remember/restore feature sometime.

**Reply by orenelbaum** [reply]
Yeah I also only found an abandoned attempt by react router.
You can read more about it here https://v5.reactrouter.com/web/guides/scroll-restoration
Apparently they just lost motivation, but in the last paragraph they are even talking about a Solid solution.

**Reply by orenelbaum** [reply]
More things I found:
https://github.com/remix-run/remix/issues/186
https://github.com/remix-run/react-router/discussions/9495
https://github.com/swup/scroll-plugin/issues/24
https://github.com/ytase/react-router-scroll/
